zookeeper-server/src/main/java/org/apache/zookeeper/server/persistence/FileTxnLog.java

```diff
@@ -289,22 +297,25 @@ public synchronized boolean append(TxnHeader hdr, Record txn, TxnDigest digest)
             logStream = new BufferedOutputStream(fos);
             oa = BinaryOutputArchive.getArchive(logStream);
             FileHeader fhdr = new FileHeader(TXNLOG_MAGIC, VERSION, dbId);
+            long dataSize = oa.getDataSize();
             fhdr.serialize(oa, "fileheader");
             // Make sure that the magic number is written before padding.
             logStream.flush();
-            filePadding.setCurrentSize(fos.getChannel().position());
+            filePosition += oa.getDataSize() - dataSize;
```

Comment:
Can you describe a bit how this calculation is correct?

Reply:
FilePosition: The position of the file which has been already flushed to file.

In this case, we initial the OutputArchive, and write the file header to it, then invoke `logStream.flush()`. 
Before writing the header, we record the dataSize, after writing the header, we use current dataSize minus the before dataSize, get the position.

Reply:
can you please add a comment in the code @horizonzy  ?